dsn_test.go

```diff
@@ -1817,3 +1818,52 @@ func TestExtractAccountName(t *testing.T) {
 		})
 	}
 }
+
+func TestUrlDecodeIfNeeded(t *testing.T) {
+	testcases := map[string]string{
+		"query_tag":             "query_tag",
+		"%24my_custom_variable": "$my_custom_variable",
+	}
+	for param, expected := range testcases {
+		t.Run(param, func(t *testing.T) {
+			decodedParam := urlDecodeIfNeeded(param)
+			assertEqualE(t, decodedParam, expected)
+		})
+	}
+}
+
+func TestUrlDecodeIfNeededE2E(t *testing.T) {
+	customVarName := "CUSTOM_VARIABLE"
+	customVarValue := "test"
+	myQueryTag := "mytag"
+	testPort, defaultPort := strconv.Atoi(os.Getenv("SNOWFLAKE_TEST_PORT"))
```

Comment:
I'm not sure I get it. Why this is a default port? I believe that the second value is an error? So maybe we can call it as a regular `err` which is a standard behaviour in Go?